### ISSUE TITLE
MA0134 - false positive in Expression<Func<Task>>

### ISSUE THREAD

**Issue by amis92** [meziantou/Meziantou.Analyzer#692]
### Version of the Meziantou.Analyzer NuGet package

2.0.139

### Rule Identifier

MA0134

### Target Framework

net8.0

### C# Language version

C# 12

### Description

For a lambda that's passed to a System.Linq.Expression parameter, it's impossible to create an async/await lambda, or consume the awaited result. It's most commonly occuring in mock-call-verification (e.g. `mock.Verify(x => x.DoAsync(It.IsAny<string>()));`).

I'm not sure if it's correct to do so, or possible, but it feels like it doesn't provide value as-is.

### Reproduction Steps

Minimal code:

```c#
using Moq;

interface IAsyncWorker {
    Task Run(int count);
}
var m = new Mock<IAsyncWorker>();
m.Verify(x => x.Run(0)); // MA0134
```
